src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFile.java

```diff
@@ -119,4 +129,51 @@ protected Type type() throws IOException, InterruptedException {
 		return this.getType();
 	}
 
+	/**
+	 * Get the correct reference of a pull request with respect to the checkout
+	 * strategy.
+	 *
+	 * This is related to JENKINS-48737. The content of file.getRef() is either the
+	 * branch name or in case of a pull request it is something like "PR-123". Since
+	 * there is no reference named like this in bitbucket it has to be changed to
+	 * the correct name. Also the configured checkout strategy for the job has to be
+	 * taken into account.
+	 *
+	 * @return the pull request reference
+	 */
+	private String getRefForPullRequest() throws IOException, InterruptedException {
+		ChangeRequestCheckoutStrategy strategy = getCheckoutStrategy();
+		String prId = ref.replace("PR-", "");
+		String source = "";
+		if (strategy != null) {
+			if (strategy == ChangeRequestCheckoutStrategy.MERGE) {
+				source = "merge";
+			} else {
+				source = "from";
+			}
+		} else {
+			throw new UnsupportedOperationException(
+					"Can not get reference for this pull request since checkout strategy can not be determined");
+		}
+		return "refs/pull-requests/" + prId + "/" + source;
+	}
+
+	@CheckForNull
+	private ChangeRequestCheckoutStrategy getCheckoutStrategy() throws IOException, InterruptedException {
+		if (this.isDirectory()) {
+			SCMRevision revision = fileSystem.getRevision();
```

Comment:
Having a look at the constructors in [line 76](https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/117/files#diff-f62992a8799c0de0f059c675c93e7708R76) and [line 80](https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/117/files#diff-f62992a8799c0de0f059c675c93e7708R80) `fileSystem` could be `null`.
I would also mark this variable `final` and maybe `@CheckForNull`.